packages/components/src/components/rating-stars/rating-stars.tsx

```diff
@@ -64,11 +68,22 @@ export class RatingStars {
   /** @deprecated in v3 in favor of kebab-case event names */
   @Event({ eventName: 'scaleChange' }) scaleChangeLegacy: EventEmitter;
 
+  connectedCallback() {
+    // make sure the deprecated props overwrite the actual ones if used
+    if (this.host.hasAttribute('max-rating')) {
+      this.max = this.maxRating;
+    }
+    if (this.host.hasAttribute('star-size')) {
+      this.size = this.starSize;
+    }
+  }
+
   // constructs the aria message for the current rating
   getRatingText() {
     const filledText = this.ariaLabelTranslation
       .replace(/\$rating/g, `${this.rating}`)
-      .replace(/\$maxRating/g, `${this.maxRating}`);
+      .replace(/\$maxRating/g, `${this.maxRating}`)
```

Comment:
so when `this.maxRating` is not updated after init (see `connectedCallback()`) should we carry `maxRating` through the component?
like isn't this  out of sync somewhere? 😅

Reply:
this one only gets effective, if the user uses a customised aria-label-translation attribute containing '$maxRating'